MiniBlog/Controllers/ArticleController.cs

```diff
@@ -2,41 +2,40 @@
 {
     using System;
     using System.Collections.Generic;
+    using System.Linq;
     using Microsoft.AspNetCore.Mvc;
     using MiniBlog.Model;
+    using MiniBlog.Service;
     using MiniBlog.Stores;
 
     [ApiController]
     [Route("[controller]")]
     public class ArticleController : ControllerBase
     {
+        private IArticleService articleService;
+
+        public ArticleController(IArticleService articleService)
+        {
+            this.articleService = articleService;
+        }
+
         [HttpGet]
         public List<Article> List()
         {
-            return ArticleStoreWillReplaceInFuture.Instance.GetAll();
+            return articleService.GetAll();
         }
 
         [HttpPost]
-        public Article Create(Article article)
+        public IActionResult Create(Article article)
         {
-            if (article.UserName != null)
-            {
-                if (!UserStoreWillReplaceInFuture.Instance.GetAll().Exists(_ => article.UserName == _.Name))
-                {
-                    UserStoreWillReplaceInFuture.Instance.Save(new User(article.UserName));
-                }
-
-                ArticleStoreWillReplaceInFuture.Instance.Save(article);
-            }
-
-            return article;
+            return Created("/articles", articleService.Create(article));
         }
 
         [HttpGet("{id}")]
         public Article GetById(Guid id)
         {
             var foundArticle =
-                ArticleStoreWillReplaceInFuture.Instance.GetAll().FirstOrDefault(article => article.Id == id);
+                articleService.GetAll().FirstOrDefault(article => article.Id == id);

```

Comment:
maybe could use your GetById method in articleService or maybe delete this method in  articleService